CHANGELOG.md

```diff
@@ -81,6 +81,8 @@ By @wumpf in [#6069](https://github.com/gfx-rs/wgpu/pull/6069), [#6099](https://
 
 - Reduce the amount of debug and trace logs emitted by wgpu-core and wgpu-hal. By @nical in [#6065](https://github.com/gfx-rs/wgpu/issues/6065)
 - `Rg11b10Float` is renamed to `Rg11b10UFloat`. By @sagudev in [#6108](https://github.com/gfx-rs/wgpu/pull/6108)
+- Removed `.global_id()` getters from wgpu objects. By @teoxoy [#6134](https://github.com/gfx-rs/wgpu/pull/6134)
+- Implement `PartialEq`, `Eq`, `Hash`, `PartialOrd` and `Ord` for wgpu resources. By @teoxoy [#6134](https://github.com/gfx-rs/wgpu/pull/6134)
```

Comment:
I think this probably needs to go in a major change bulitin explaining the migration.

Reply:
What *is* the migration? How are applications currently using `global_id()` going to get something with equivalent properties (reliably distinguishes resources, but *does not* keep the resource alive) without additional overhead? I use these a lot to ensure resources that depend on other resources (texture views, bind groups) are up-to-date.

Hm, I suppose applications could generate their own IDs and maybe even carry them in a wrapper implementing  `Deref` and get the same results. Still, I suspect something useful is being lost here.

Reply:
The main problem is after this PR we don't have such an ID anymore, so would need to generate it with an atomic on all platforms. 

If this is indeed required functionality we can absolutely have the wgpu layer do that, though they're might be other ways to solve that problem now that things impl PartialEq, Ord, and Hash

Reply:
Well, I wouldn't want to demand that wgpu produce something it doesn't use, so probably, application-generated IDs (or, in my case, perhaps merely generation numbers) are a reasonable solution for those applications that need them.

(Sorry for jumping in late and complaining a bit.)

Reply:
Sounds reasonable - thanks for the feedback, part of why I was unsure about this is I didn't exactly know if we were covering the use cases of people.

Reply:
FWIW, here’s what the migration looks like for me, adding a wrapper with IDs to the places it was needed: https://github.com/kpreid/all-is-cubes/commit/87fb899aa0ef22489211257c4c851a3a7b9eed58. I tested it against wgpu trunk.